kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/RequestFilter.java

```diff
@@ -52,9 +52,42 @@ default boolean shouldHandleRequest(ApiKeys apiKey, short apiVersion) {
      *         request to be forwarded.
      * @see io.kroxylicious.proxy.filter Creating Filter Result objects
      * @see io.kroxylicious.proxy.filter Thread Safety
+     * @deprecated implement {@link #onRequest(ApiKeys, short, RequestHeaderData, ApiMessage, FilterContext)} instead.
      */
-    CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey,
-                                                   RequestHeaderData header,
-                                                   ApiMessage request,
-                                                   FilterContext context);
+    @Deprecated(forRemoval = true, since = "0.19.0")
+    default CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey,
```

Comment:
There's a bit of a compromise going on here.  My intent is that the code of existing authors who have implemented the RequestFilter and ResponseFilter continues to work unchanged.

* the default implementation on the new method achieves this for _existing compiled implementations_.  `MyRequestFilter.class` will continue to work as it does today and won't need to be changed / recompiled.
* if the filter author is recompiling existing code against 0.19.0, their code will compile providing they weren't relying on functional interfaces.  `RequestFilter` and `ResponseFilter` were _undeclared_ `@FuncionalInterface` so the filter author could exploit that in their code (lambda expressions).  If they have done that, they'll need to change code.   I don't think I can avoid this.

Reply:
sounds good, it might be nice to also log something if we find an implementation that implements the old method

Here's a [PR](https://github.com/k-wall/kroxylicious/pull/121) adding logs